sound/soc/sof/pcm.c

```diff
@@ -673,6 +673,18 @@ static int sof_pcm_ack(struct snd_soc_component *component,
 	return snd_sof_pcm_platform_ack(sdev, substream);
 }
 
+static snd_pcm_sframes_t sof_pcm_delay(struct snd_soc_component *component,
+				       struct snd_pcm_substream *substream)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	const struct sof_ipc_pcm_ops *pcm_ops = sdev->ipc->ops->pcm;
+
```

Comment:
for the open and close, we have this test:

	/* nothing to do for BE */
	if (rtd->dai_link->no_pcm)
		return 0;


is this not needed here?

Reply:
we check no_pcm since the following ops will be called by both fe & be.  We don't check it in ack function since it is not in such case. BTW we also don't need to check no_pcm in sof_pcm_pointer with the same reason.

````c
		rtd->ops.open		= dpcm_fe_dai_open;
		rtd->ops.hw_params	= dpcm_fe_dai_hw_params;
		rtd->ops.prepare	= dpcm_fe_dai_prepare;
		rtd->ops.trigger	= dpcm_fe_dai_trigger;
		rtd->ops.hw_free	= dpcm_fe_dai_hw_free;
		rtd->ops.close		= dpcm_fe_dai_close;
````